viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
+# from tkinter.tix import InputOnly
+
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+        # validate input
+    new_dict = {}
+    new_dict["title"] = title
+    new_dict["genre"] = genre
+    new_dict["rating"] = rating
+    # if not new_dict["key"]
+    # for key in new_dict:
+    #     if not new_dict[key]:
+    #         return None
+    return new_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            i = user_data["watchlist"].index(movie)
+            user_data["watchlist"].pop(i)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
+
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+
+    rating_list=[]
+    for movie in user_data["watched"]:
+        if movie["rating"]> 0:

```

Comment:
Nice job not calculating zero ratings. I was initially concerned that they might have been not accounted for in the average, but you took care of it using `len`!